docs/reference/stdlib.md

```diff
@@ -194,111 +187,121 @@ The following constants are globally available in a Nextflow script for workflow
 
-  `glob`
+  `glob: boolean`
   : When `true`, interprets characters `*`, `?`, `[]` and `{}` as glob wildcards, otherwise handles them as normal characters (default: `true`)
 
-  `hidden`
+  `hidden: boolean`
   : When `true`, includes hidden files in the resulting paths (default: `false`)
 
-  `maxDepth`
+  `maxDepth: int`
   : Maximum number of directory levels to visit (default: *no limit*)
 
-  `type`
+  `type: String`
   : Type of paths returned, can be `'file'`, `'dir'` or `'any'` (default: `'file'`)
 
 : See also: {ref}`Channel.fromPath <channel-path>`.
 
-`files( filePattern, [options] )`
-: Convenience method for `file()` that always returns a list.
+`files( filePattern: String, [options] ) -> List<Path>`
+: Convenience function for `file()` that always returns a list.
 
-`groupKey( key, size )`
+`groupKey( key, size: int ) -> GroupKey`
 : Create a grouping key to use with the {ref}`operator-grouptuple` operator.
 
-`multiMapCriteria( closure )`
+`multiMapCriteria( criteria: Closure ) -> Closure`
 : Create a multi-map criteria to use with the {ref}`operator-multiMap` operator.
 
-`sendMail( params )`
-: Send an email. See {ref}`mail-page`.
+`sendMail( [options] )`
+: Send an email. See {ref}`mail-page` for more information.
+
+`sleep( milliseconds: long )`
+: Sleep for the given number of milliseconds.
 
-`tuple( collection )`
+`tuple( collection: List ) -> ArrayTuple`
 : Create a tuple object from the given collection.
 
-`tuple( ... args )`
+`tuple( args... ) -> ArrayTuple`
 : Create a tuple object from the given arguments.
 
-`workflow.onComplete( closure )`
-: Define an action to take when the workflow completes (whether successful or not). Refer to the `workflow` implicit variable to see which additional properties are available in the completion handler.
-
-`workflow.onError( closure )`
-: Define an action to take if the workflow is terminated due to a runtime error or task failure. Refer to the `workflow` implicit variable to see which additional properties are available in the error handler.
-
-(stdlib-default-imports)=
+## Groovy and Java classes
 
-## Default imports
-
-The following classes are imported by default in Nextflow scripts:
+Any Groovy or Java class can be used in a Nextflow script. The following classes are imported by default:
 
+- `groovy.lang.*`
+- `groovy.util.*`
 - `java.io.*`
 - `java.lang.*`
 - `java.math.BigDecimal`
 - `java.math.BigInteger`
 - `java.net.*`
-- `java.nio.file.Path`
 - `java.util.*`
-- `groovy.lang.*`
-- `groovy.util.*`
 
-## Channel
+All other classes must be referenced by their fully-qualified name:
 
-The `Channel` class provides the channel factory methods. See {ref}`channel-factory` for more information.
+```nextflow
+def vals = [1, 2, 3]
+println groovy.json.JsonOutput.toJson(vals)
+```
 
-(stdlib-types-duration)=
+The following sections describe the standard types provided by Nextflow.
 
-## Duration
+## Bag\<E\>
```

Comment:
This is an internal structure, not sure it should be documented as public API

Reply:
It will need to be part of the public API to represent things like the values of a map or the return type of the `collect` operator

The standard collections are:
- Bag: unordered, duplicates allowed
- List: ordered, duplicates allowed
- Set: unordered, duplicates not allowed